src/roiextractors/segmentationextractor.py

```diff
@@ -319,3 +323,120 @@ def write_segmentation(segmentation_extractor, save_path, overwrite=False):
         """
         raise NotImplementedError
+
+
+class FrameSliceSegmentationExtractor(SegmentationExtractor):
+    """
+    Class to get a lazy frame slice.
+
+    Do not use this class directly but use `.frame_slice(...)`
+    """
+
+    extractor_name = "FrameSliceSegmentationExtractor"
+    is_writable = True
+
+    def __init__(
+        self,
+        parent_segmentation: SegmentationExtractor,
+        start_frame: Optional[int] = None,
+        end_frame: Optional[int] = None,
+    ):
+        self._parent_segmentation = parent_segmentation
+        self._start_frame = start_frame or 0
+        self._end_frame = end_frame or self._parent_segmentation.get_num_frames()
+        self._num_frames = self._end_frame - self._start_frame
+
+        self._image_masks = self._parent_segmentation._image_masks
+
+        parent_size = self._parent_segmentation.get_num_frames()
+        if start_frame is None:
+            start_frame = 0
+        else:
+            assert 0 <= start_frame < parent_size
+        if end_frame is None:
+            end_frame = parent_size
+        else:
+            assert 0 < end_frame <= parent_size
+        assert end_frame > start_frame, "'start_frame' must be smaller than 'end_frame'!"
+
+        super().__init__()
+        if getattr(self._parent_segmentation, "_times") is not None:
+            self._times = self._parent_segmentation._times[start_frame:end_frame]
+
+    def get_accepted_list(self) -> list:
+        return self._parent_segmentation.get_accepted_list()
+
+    def get_rejected_list(self) -> list:
+        return self._parent_segmentation.get_rejected_list()
+
+    def get_traces(
+        self,
+        roi_ids: Optional[Iterable[int]] = None,
+        start_frame: Optional[int] = None,
+        end_frame: Optional[int] = None,
+        name: str = "raw",
+    ) -> np.ndarray:
+        start_frame = min(start_frame or 0, self._num_frames)
+        end_frame = min(end_frame or self._num_frames, self._num_frames)
+        return self._parent_segmentation.get_traces(
+            roi_ids=roi_ids,
+            start_frame=start_frame + self._start_frame,
+            end_frame=end_frame + self._start_frame,
+            name=name,
+        )
+
+    def get_traces_dict(self):
+        for trace in self._parent_segmentation.get_traces_dict().values():
+            if trace is not None and len(trace.shape) > 0:
+                if not isinstance(trace, np.ndarray):
+                    raise NotImplementedError(
+                        "The `get_traces_dict` method for SubFrameSegementations does not yet support underlying trace "
+                        "types other than numpy arrays (which includes memory maps)."
+                    )
+
+        def _safe_subtrace(trace: Optional[np.ndarray], start: int, end: int):
+            if trace is None and len(trace.shape) > 0:
+                return None
+            return trace[start:end, :]
+
+        return dict(
```

Comment:
What do you think of going over the dict of the parent and using the names instead. That would make this shorter and will be future proof if we add one more type of trace (trace_name) in the future.

Reply:
Done

Reply:
Great.